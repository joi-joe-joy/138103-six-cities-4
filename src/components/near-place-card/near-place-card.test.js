import React from "react";
import renderer from "react-test-renderer";
import NearPlaceCard from "./near-place-card.jsx";

const offer = {
  id: 2,
  title: `Wood and stone place`,
  premium: false,
  pictures: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
    `img/room.jpg`,
    `img/studio-01.jpg`,
    `img/studio-photos.jpg`
  ],
  price: 180,
  rating: 4.3,
  type: `hotel`,
  bedrooms: 2,
  amenities: [`Indoor fireplace`, `Kitchen`, `Wifi`, `Washer`],
  maxGuestsNumber: `Max 2 adults, 2 children`,
  description: `Newly renovated quiet cozy haven in the middle of the hustle and bustle of New York City with breathtaking views of the Manhattan skyline from every window.`,
  host: {
    photo: `img/avatar-max.jpg`,
    name: `Maxim Doff`,
    super: false
  },
  coordinations: [52.369553943508, 4.85309666406198],
  reviews: [],
  nearOffers: []
};

describe(`Render NearPlaceCard`, () => {
  it(`Render NearPlaceCard correctly`, () => {
    const tree = renderer
      .create(
          <NearPlaceCard
            key={1}
            offer={offer}
            onCardHover={()=>{}}
            onBookmarkClick={()=>{}}
            onCardClick={()=>{}}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Render NearPlaceCard correctly with classNames`, () => {
    const tree = renderer
      .create(
          <NearPlaceCard
            key={1}
            offer={offer}
            onCardHover={()=>{}}
            onBookmarkClick={()=>{}}
            onCardClick={()=>{}}
            classNames={{
              card: `test`,
              imgWrap: `test`
            }}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
