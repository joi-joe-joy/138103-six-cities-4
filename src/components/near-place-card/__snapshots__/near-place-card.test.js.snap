// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NearPlaceCard Render NearPlaceCard correctly 1`] = `
<article
  className="place-card near-places__card "
  onMouseOver={[Function]}
>
  <div
    className="near-places__image-wrapper  place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Wood and stone place"
        className="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          180
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        onClick={[Function]}
        type="button"
      >
        <SvgrURL
          height="18"
          width="17"
        />
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        onClick={[Function]}
      >
        Wood and stone place
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Hotel
    </p>
  </div>
</article>
`;

exports[`Render NearPlaceCard Render NearPlaceCard correctly with classNames 1`] = `
<article
  className="place-card near-places__card test"
  onMouseOver={[Function]}
>
  <div
    className="near-places__image-wrapper test place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Wood and stone place"
        className="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          180
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        onClick={[Function]}
        type="button"
      >
        <SvgrURL
          height="18"
          width="17"
        />
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        onClick={[Function]}
      >
        Wood and stone place
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Hotel
    </p>
  </div>
</article>
`;
